src/tr2/game/objects/general/general.c

```diff
@@ -0,0 +1,14 @@
+#include "game/objects/general/general.h"
+
+#include "game/objects/common.h"
+#include "global/funcs.h"
+
+void General_Setup(void)
+{
+    OBJECT *const obj = Object_GetObject(O_GENERAL);
```

Comment:
I was curious about what this was and why it's classed as a water creature - it's actually the submarine at the beginning of 40 Fathoms. The water flag must be to allow the SFX to play when it hits the seabed.
Is it worth renaming? Or perhaps once `GeneralControl` is ported to confirm what it does.

Reply:
From TOMB21:
```
    /* GeneralControl is used for anything that just gets activated, moves and
     * stops again */
```
`water_creature` is used scarcely, so its effects on how the object behaves are limited. I'm good with renaming, I just don't have a proposal.